### ISSUE TITLE
Feature request: Java SDK

### ISSUE THREAD

**Issue by gunnarmorling** [slatedb/slatedb#368]
See the initial discussion in #243. It would be great to have Java bindings for SlateDB, for instance using the new Panama FFI.

**Comment by criccomini** [comment]
cc @dmvk It sounded like you might be working with SlateDB through Java. Do you have an FFI binding you're working on? No biggy if not, but wanted to check.

**Comment by criccomini** [comment]
Ah, nevermind. I just saw your comment on #243.

**Comment by criccomini** [comment]
Seems https://github.com/jni-rs/jni-rs is another option..

**Comment by gunnarmorling** [comment]
Yeah, gonna take a look at that one, too. Maybe it is helping with that ABI
issue discussed in #243.

**Comment by criccomini** [comment]
Yea, it's unfortunate that the `object_store` crate doesn't have a Java binding (that I know of). If it did, we could write a JNI binding for the Java interface, and on the Rust side, write an `ObjectStore` implementation that wrapped the JNI binding. This would be an argument to use OpenDAL since it already has bindings for many languages. /cc @Xuanwo

**Comment by pditommaso** [comment]
I wanted to give a try to this over the weekend. I put together Java client starting from the Go binding. 

You can find it [here](https://github.com/pditommaso/slatedb-java). Happy to contribute back to the community.

**Comment by criccomini** [comment]
@pditommaso this is awesome! Thanks for doing this. If you're up for it, we'd definitely want to have it as a module under `slatedb-java` in our repo. I can handle setting up maven central publication.

**Comment by pditommaso** [comment]
You are welcome. I'm very excited by this project! I'd be very happy to contribute it in the main repo. I'll open a PR in the following days.

**Comment by criccomini** [comment]
@pditommaso awesome! Looking forward to the PR. 😀

**Comment by pditommaso** [comment]
Can I be added as a contributor to the project to submit the PR or do you prefer that I create a fork ?

**Comment by criccomini** [comment]
> do you prefer that I create a fork

Please create a fork and submit a PR. Becoming a committer is a much longer process (see [CONTRIBUTING.md](https://github.com/slatedb/slatedb/blob/main/CONTRIBUTING.md)). Note that this means any future changes will need to be reviewed, merged, and released by a committer. (If you're uncomfortable with that, you might want to reconsider contributing the PR.) Thus far, we've been treating the other bindings as alpha/beta, and merging quite aggressively, so I don't think it'll be an issue.

**Comment by pditommaso** [comment]
That's fair

**Comment by criccomini** [comment]
The FFI bindings are now available on Maven Central:

https://central.sonatype.com/artifact/io.slatedb/slatedb

The JAR includes libslatedb_c.{so,dylib,dll} in the JAR and autoload so no local build is required.

cc @em3s
